From 25a9bac9f0a9346e9d53712d4d25186bc5306de6 Mon Sep 17 00:00:00 2001
From: Kamal Mostafa <kamal@canonical.com>
Date: Thu, 6 Sep 2012 14:28:29 -0700
Subject: [PATCH] UBUNTU: SAUCE: input: Cypress PS/2 Trackpad move
 PSMOUSE_CYPRESS enum

Move PSMOUSE_CYPRESS to the end of the enum, since awful
src/eventcomm.c in x-x-i-s actually depends on hardcoded values
of the model numbers (including MODEL_ALPS).  *sigh*

BugLink: http://launchpad.net/bugs/1041594

Signed-off-by: Kamal Mostafa <kamal@canonical.com>
Signed-off-by: Tim Gardner <tim.gardner@canonical.com>
---
 drivers/input/mouse/psmouse.h | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/input/mouse/psmouse.h b/drivers/input/mouse/psmouse.h
index ed788d0..148286b 100644
--- a/drivers/input/mouse/psmouse.h
+++ b/drivers/input/mouse/psmouse.h
@@ -79,6 +79,9 @@ struct psmouse {
 };
 
 enum psmouse_type {
+	/* IMPORTANT! src/eventcomm.c in x-x-i-s uses the hard-coded values
+	 * of some of these (awful!), so never insert new entries to the
+	 * middle of the list!  */
 	PSMOUSE_NONE,
 	PSMOUSE_PS2,
 	PSMOUSE_PS2PP,
@@ -87,7 +90,6 @@ enum psmouse_type {
 	PSMOUSE_IMPS,
 	PSMOUSE_IMEX,
 	PSMOUSE_SYNAPTICS,
-	PSMOUSE_CYPRESS,
 	PSMOUSE_ALPS,
 	PSMOUSE_LIFEBOOK,
 	PSMOUSE_TRACKPOINT,
@@ -97,6 +99,7 @@ enum psmouse_type {
 	PSMOUSE_ELANTECH,
 	PSMOUSE_FSP,
 	PSMOUSE_SYNAPTICS_RELATIVE,
+	PSMOUSE_CYPRESS,
 	PSMOUSE_AUTO		/* This one should always be last */
 };
 
-- 
1.8.0

